k4FWCore/include/k4FWCore/MetaDataHandle.h

```diff
@@ -0,0 +1,108 @@
+#ifndef K4FWCORE_METADATAHANDLE_H
+#define K4FWCORE_METADATAHANDLE_H
+
+// GAUDI
+#include "GaudiAlg/GaudiAlgorithm.h"
+
+#include "podio/GenericParameters.h"
+#include "k4FWCore/PodioDataSvc.h"
+
+#include "GaudiKernel/MsgStream.h"
+
+#include <type_traits>
+
+namespace k4FWCore {
+
+static std::string metaDataHandleDescriptor(const std::string& collectionName, const std::string& key) {
+  return collectionName + "__" + key;
+}
+
+} // namespace
+
+template <typename T> class MetaDataHandle {
+public:
+  MetaDataHandle();
+  MetaDataHandle(const std::string& descriptor, Gaudi::DataHandle::Mode a);
+  MetaDataHandle(const Gaudi::DataHandle& handle, const std::string& descriptor, Gaudi::DataHandle::Mode a);
+  ~MetaDataHandle();
+
+  const T get();
+  void put(T);
+  
+private:
+  std::string fullDescriptor();
+
+private:
+  ServiceHandle<IDataProviderSvc> m_eds;
+  std::string                     m_descriptor;
+  PodioDataSvc*                   m_podio_data_service{nullptr};
+  const Gaudi::DataHandle*        m_dataHandle{nullptr}; // holds the identifier in case we do collection metadata
+  Gaudi::DataHandle::Mode         m_mode;
+
+};
+
+template <typename T> MetaDataHandle<T>::~MetaDataHandle() {
+}
+
+//---------------------------------------------------------------------------
+template <typename T>
+MetaDataHandle<T>::MetaDataHandle(const std::string& descriptor, Gaudi::DataHandle::Mode a)
+: m_eds("EventDataSvc", "DataHandle"), m_descriptor(descriptor), m_mode(a) {
+
+  StatusCode    sc = m_eds.retrieve();
+  m_podio_data_service = dynamic_cast<PodioDataSvc*>(m_eds.get()); 
+  if (nullptr == m_podio_data_service) {
+    std::cout << "ERROR: MetaDataHandles require the PodioDataSvc" << std::endl;
+  }
+}
+
+//---------------------------------------------------------------------------
+template <typename T>
+MetaDataHandle<T>::MetaDataHandle(const Gaudi::DataHandle& handle, const std::string& descriptor, Gaudi::DataHandle::Mode a)
+: m_eds("EventDataSvc", "DataHandle"), m_descriptor(descriptor), m_dataHandle(&handle), m_mode(a) {
+
+  StatusCode    sc = m_eds.retrieve();
+  m_podio_data_service = dynamic_cast<PodioDataSvc*>(m_eds.get());
+  if (nullptr == m_podio_data_service) {
+    std::cout << "ERROR: MetaDataHandles require the PodioDataSvc" << std::endl;
```

Comment:
This line and the same line in the constructor above print these errors during compilation. This could be confusing for someone and we shouldn't have them but I'm not sure if there is a way other than removing the lines completely. This happens when running `genconf` here: https://github.com/key4hep/k4FWCore/blob/master/test/k4FWCoreTest/CMakeLists.txt#L11

Reply:
I know it is not pretty, but I worked around the problem in a couple of cases in Gaudi like this: https://gitlab.cern.ch/gaudi/Gaudi/-/blob/v36r14/GaudiExamples/src/AlgTools/MyGaudiTool.cpp#L79-80